grpc/py-cubic/cubic/client.py

```diff
@@ -0,0 +1,142 @@
+
+class Client(object):
+    """ A client for interacting with Cobalt's Cubic GRPC API."""
+
+    def __init__(self, serverAddress, secure=True,
+                serverCertificate=None,
+                clientCertificate=None,
+                clientKey=None,
+                bufferSize=8192):
+        """  Creates a new cubic Client object.
+
+        Args:
+            serviceAddress: host:port of where cubic server is running
+
+            secure: If set to true, channel with transport security is used.
+                    Otherwise, an insecure grpc channel is used. Default is True.
+
+            serverCertificate:  PEM certificate as byte string which is used as a  
+                                root certificate that can validate the certificate 
+                                presented by the server we are connecting to. Use this 
+                                when connecting to an instance of cubic server that is 
+                                using a self-signed certificate.
+
+            clientCertificate:  PEM certificate as bytes string presented by this Client when
+                                connecting to a server. Use this when setting up mutually 
+                                authenticated TLS. The clientKey must also be provided.
+
+            clientKey:  PEM key as byte string presented by this Client when
+                        connecting to a server. Use this when setting up mutually 
+                        authenticated TLS. The clientKey must also be provided.
+
+            bufferSize: Bytes of audio sent to cubic server in each chunk if using streaming
+                        recognition. Default is 8192 bytes.
+        """
+
+        self.serverAddress = serverAddress
+        self.secure = secure
+        self.bufferSize = bufferSize
+        
+        if self.bufferSize <= 0:
+            raise ValueError('buffer size must be greater than 0')
+
+        if secure:
+            # using a TLS endpoint with optional certificates for mutual authentication
+            if clientCertificate is not None and clientKey is None:
+                raise ValueError("client key has not been provided")
+            if clientKey is not None and clientCertificate is None:
+                raise ValueError("client certificate has not been provided")
+            self._creds = grpc.ssl_channel_credentials(
+                root_certificates=serverCertificate, 
+                private_key=clientKey,
+                certificate_chain=clientCertificate)
+            self._channel = grpc.secure_channel(serverAddress, self._creds)
+        else:
+            self._channel = grpc.insecure_channel(serverAddress)
+
+        self._client = CubicStub(self._channel)
+
+    def __del__(self):
+        """ Closes and cleans up after Client. """
+        try:
+            self._channel.close()
+        except AttributeError:
+            # client wasn't 
```

Comment:
client wasn't what?

Reply:
oops, was supposed to be something along the lines of ` # client was't fully instantiated, no channel to close `